java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -1020,12 +1021,16 @@ public void clearStack(Tree tree) {
   }
 
   private void setSymbolicValueOnFields(MethodInvocationTree tree) {
-    if (isLocalMethodInvocation(tree) || isProvidingThisAsArgument(tree) || THREAD_SLEEP_MATCHER.matches(tree)) {
-      resetFieldValues();
+    boolean threadSleepMatch = THREAD_SLEEP_MATCHER.matches(tree);
+    if (isLocalMethodInvocation(tree) || isProvidingThisAsArgument(tree) || threadSleepMatch) {
+      resetFieldValues(tree.symbol().isStatic() && !threadSleepMatch);
```

Comment:
If {{this}} is provided as argument of a static method, non-static fields should also be reset, which won't be the case here.